src/common/meta/src/key.rs

```diff
@@ -926,22 +924,35 @@ impl TableMetadataManager {
     }
 
     /// Updates table info and returns an error if different metadata exists.
+    /// And cascade-ly update all redundant table options for each region
+    /// if region_distribution is present.
     pub async fn update_table_info(
         &self,
         current_table_info_value: &DeserializedValueWithBytes<TableInfoValue>,
+        region_distribution: Option<RegionDistribution>,
         new_table_info: RawTableInfo,
     ) -> Result<()> {
         let table_id = current_table_info_value.table_info.ident.table_id;
-
         let new_table_info_value = current_table_info_value.update(new_table_info);
 
         // Updates table info.
         let (update_table_info_txn, on_update_table_info_failure) = self
             .table_info_manager()
             .build_update_txn(table_id, current_table_info_value, &new_table_info_value)?;
 
-        let mut r = self.kv_backend.txn(update_table_info_txn).await?;
+        let txn = if let Some(region_distribution) = region_distribution {
```

Comment:
make sure `region_distribution.len() != 0` is true

Reply:
Checked here https://github.com/GreptimeTeam/greptimedb/blob/3a492fecf82d31e00d340f3baf5bd6683e5ac731/src/common/meta/src/key/datanode_table.rs#L223